ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto

```diff
@@ -223,14 +223,22 @@ message DisclosedContract {
   // Required
   oneof arguments {
     // The contract arguments as typed Record
-    Record create_arguments = 3;
+    // Deprecated in favor of `contract_create_payload`
+    Record create_arguments = 3 [deprecated = true];
 
     // The contract arguments specified using an opaque blob extracted from the ``create_arguments_blob`` field
     // of a ``com.daml.ledger.api.v1.CreatedEvent``.
-    google.protobuf.Any create_arguments_blob = 5;
+    // Deprecated in favor of `contract_create_payload`
+    google.protobuf.Any create_arguments_blob = 5 [deprecated = true];
   }
 
   // The contract metadata from the create event.
+  // Deprecated in favor of `contract_create_payload`
+  // Optional
+  ContractMetadata metadata = 4 [deprecated = true];
+
+  // Opaque byte string containing the complete payload required by the Daml engine
+  // to reconstruct a contract not known to the receiving participant.
   // Required
```

Comment:
This one `required` doesn't make much sense if the above are still deprecated but supported, does it? Or how do we envision to support both? If there is a transition period in which we support the deprecated ones, the this one would be naturally `Optional` no? 

Unless I'm wrong in my assessment, I'm fine to leave it like this, provided that we completely remove the deprecated ones before the release cut.

Reply:
It works both ways for me depending on how do we understand the interplay with the deprecated fields

- if we consider an idealized interface where all deprecated fields are "invisible", this field becomes Required
- if OTOH, we consider the interface literally and emphasize what the code behind it needs to do, the field is clearly Optional

I tend to interpret it in the first way, thus driving the client implementations towards the idealized future interface